src/System.Private.CoreLib/src/System/Object.CoreCLR.cs

```diff
@@ -16,7 +26,33 @@ public partial class Object
         // object.  This is always a shallow copy of the instance. The method is protected
         // so that other object may only call this method on themselves.  It is intended to
         // support the ICloneable interface.
-        [MethodImpl(MethodImplOptions.InternalCall)]
-        protected extern object MemberwiseClone();
+        protected unsafe object MemberwiseClone()
+        {
+            object clone = this;
+
+            AllocateClone(JitHelpers.GetObjectHandleOnStack(ref clone));
```

Comment:
Assuming I'm reading this correctly, if `this` is _null_, the call to `GetObjetHandleOnStack` will no-op, and the call to `GetMethodTable` below will NRE. Sound about right?

Reply:
Yep.

Reply:
Added comment.